timesketch/lib/analyzers/sigma_tagger.py

```diff
@@ -71,64 +46,54 @@ def run(self):
-                        try:
-                            rule_file_content = rule_file.read()
-                            parser = sigma_collection.SigmaCollectionParser(
-                                rule_file_content, self.sigma_config, None)
-                            parsed_sigma_rules = parser.generate(sigma_backend)
-                        except NotImplementedError as exception:
-                            logger.error(
-                                'Error generating rule in file {0:s}: {1!s}'
-                                .format(rule_file_path, exception))
-                            continue
-
-                        for sigma_rule in parsed_sigma_rules:
-                            try:
-                                sigma_rule_counter += 1
-                                logger.info(
-                                    '[sigma] Generated query {0:s}'
-                                    .format(sigma_rule))
-                                tagged_events_counter = self.run_sigma_rule(
-                                    sigma_rule, tag_name)
-                                tags_applied[tag_name] += tagged_events_counter
-                            except elasticsearch.TransportError as e:
-                                logger.error(
-                                    'Timeout generating rule in file {0:s}: '
-                                    '{1!s} waiting for 10 seconds'.format(
-                                        rule_file_path, e), exc_info=True)
-                                time.sleep(10) # waiting 10 seconds
+        try:
+            rules_path = ts_sigma_lib.get_sigma_rules_path()
+
+        except ValueError:
+            logger.error('OS Error, unable to get the path to the Sigma rules',
+                         exc_info=True)
+            return False
+
+        sigma_rules = ts_sigma_lib.get_sigma_rules(rules_path)
+
+        if sigma_rules is None:
+            logger.error('error rule is none')
+
+        problem_string = 'Problematic rules:\n'
+
+        for rule in sigma_rules:
+            if rule is None:
+                logger.error('error rule is none')
+                continue
+            tags_applied[rule.get('file_name')] = 0
+            logger.debug("Rule : {0:s} with query {1:s}".format(
+                rule.get('id'), rule.get('file_name')))
+            try:
+                sigma_rule_counter += 1
+                tagged_events_counter = self.run_sigma_rule(
+                    rule.get('es_query'), rule.get('file_name'))
+                tags_applied[rule.get('file_name')] += tagged_events_counter
+            except elasticsearch.TransportError as e:
+                logger.error(
+                    'Timeout executing search for {0:s}: '
+                    '{1!s} waiting for 10 seconds'.format(
+                        rule.get('file_name'), e), exc_info=True)
+                time.sleep(10) # waiting 10 seconds
```

Comment:
can you add a comment about why you are sleeping for 10 seconds here?

and why 10? why not 1, or 2?

Reply:
The 10 seconds was a compromise I tries with my local dev machine, lower numbers where in most cases not enough for ES to recover.